src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -170,59 +207,77 @@ public ComplaintResponse createComplaintResponseRepresentingLock(Complaint compl
+
+        if (updatedComplaintResponse.getComplaint().isAccepted() == null) {
+            throw new IllegalArgumentException("You need to either accept or reject a complaint");
         }
-        Optional<ComplaintResponse> complaintResponseFromDatabaseOptional = complaintResponseRepository.findById(updatedComplaintResponse.getId());
-        if (complaintResponseFromDatabaseOptional.isEmpty()) {
-            throw new IllegalArgumentException("The complaint response was not found in the database");
+
+        validateResponseTextLimit(updatedComplaintResponse.getResponseText(), originalComplaint);
+
+        originalComplaint.setAccepted(updatedComplaintResponse.getComplaint().isAccepted()); // accepted or denied
+        originalComplaint = complaintRepository.save(originalComplaint);
+
+        complaintResponseFromDatabase.setSubmittedTime(ZonedDateTime.now());
+        complaintResponseFromDatabase.setResponseText(updatedComplaintResponse.getResponseText());
+        complaintResponseFromDatabase.setComplaint(originalComplaint);
+        complaintResponseFromDatabase.setReviewer(user);
+        return complaintResponseRepository.save(complaintResponseFromDatabase);
+    }
+
+    private void validateComplaintResponseId(Long complaintResponseId) {
+        if (complaintResponseId == null) {
+            throw new IllegalArgumentException("The complaint response needs to have an id");
         }
-        ComplaintResponse emptyComplaintResponseFromDatabase = complaintResponseFromDatabaseOptional.get();
-        if (emptyComplaintResponseFromDatabase.getSubmittedTime() != null || emptyComplaintResponseFromDatabase.getResponseText() != null) {
+    }
+
+    private ComplaintResponse fetchComplaintResponseOrThrow(Long complaintResponseId) {
+        return complaintResponseRepository.findById(complaintResponseId).orElseThrow(() -> new IllegalArgumentException("The complaint response was not found in the database"));
+    }
+
+    private void validateComplaintResponseEmpty(ComplaintResponse complaintResponse) {
+        if (complaintResponse.getSubmittedTime() != null || complaintResponse.getResponseText() != null) {
             throw new IllegalArgumentException("The complaint response is not empty");
         }
-        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(emptyComplaintResponseFromDatabase.getComplaint().getId());
-        if (originalComplaintOptional.isEmpty()) {
-            throw new IllegalArgumentException("The complaint was not found in the database");
-        }
-        Complaint originalComplaint = originalComplaintOptional.get();
+    }
+
+    private Complaint fetchOriginalComplaintOrThrow(ComplaintResponse complaintResponse) {
+        return complaintRepository.findByIdWithEagerAssessor(complaintResponse.getComplaint().getId())
+                .orElseThrow(() -> new IllegalArgumentException("The complaint was not found in the database"));
```

Comment:
Should be moved to the repository as a default implementation. Check out other repositories for similar implementations.